File: acido.java
import java.util.Scanner;

public class acido {
	
	private static boolean ispossible(String vet, int i, int j) 
	{
		if( i < 0 || j >= vet.length() )
			return false;
		if( vet.charAt(i) == 'C' && vet.charAt(j) == 'F' || 
			vet.charAt(i) == 'F' && vet.charAt(j) == 'C' )
			return true;
		if( vet.charAt(i) == 'S' && vet.charAt(j) == 'B' || 
			vet.charAt(i) == 'B' && vet.charAt(j) == 'S' )
			return true;
		
		return false;	
	}
	
	/**
	 * @param args
	 */
	/**
	 * @param args
	 */
	public static void main(String... args) {
		Scanner scn = new Scanner(System.in);
		
		String str = scn.next();
		int contagem = 0;
		
		while( true ) {
			int maior = 0, k = 0, indMaior = -1;
			
			for( int i = 1; i < str.length(); ++i ) {
				int x = i-1;
				int y = i;
				k = 0;
				
				while( ispossible(str, x, y) ) {
					x--; y++; k++;
				}
				
				if( k > maior ) {
					indMaior = i;
					maior = k;
				}
			}
			if( maior != 0 ) {
				contagem += maior;
				StringBuilder temp = new StringBuilder("");
				for( int i = 0; i < str.length(); ++i )
					if( i < (indMaior-maior) || i >= (indMaior+maior) )
						temp.append(String.valueOf(str.charAt(i)));
				
				str = temp.toString();
			} else 
				break;
		}
		System.out.println(contagem);
	}

}

File: acido.cpp
#include <cstdio>
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

const int MAX = 300;

string vet;

inline bool ispossible(int i, int j)
{
    if( i < 0 || j >= (int) vet.size() ) return false;
    if( (vet[i] == 'C' && vet[j] == 'F') || (vet[i] == 'F' && vet[j] == 'C') ) return true;
    if( (vet[i] == 'S' && vet[j] == 'B') || (vet[i] == 'B' && vet[j] == 'S') ) return true;
    return false;
}

int main()
{

    while( cin >> vet ) {
        int contagem = 0;
        while( true ) {
            int maior = 0, k = 0, indMaior = -1;

            for( int i = 1; i < (int)vet.size(); ++i ) {
                int x = i-1;
                int y = i;
                k = 0;

                while( ispossible(x, y) ) {
                    x--; y++; k++;
                }

                if( k > maior ) {
                    indMaior = i;
                    maior = k;
                }
            }
            if( maior ) {
                contagem += maior;
                vet.erase(vet.begin()+(indMaior-maior), vet.begin()+(indMaior+maior));
            } else
                break;

        }
        printf("%d\n", contagem);

    }






    return 0;
}
